### PR TITLE
Cleaned up tests to use more narrow assertions

### PR DESCRIPTION
**Pull request by mjriley**
## Technical Summary
This is followup work from https://github.com/dimagi/commcare-hq/pull/34082, where, because that PR modified the `UserData` object to also contain default properties from the domain, any tests that asserted against the entire `UserData` object failed. Several of these failing tests could instead narrowly assert against only the properties that they are testing against. I also moved at least one test around as it didn't seem to apply to `TestMobileUserBulkUpload`

### Safety story
This only affects the tests.  While this does reduce the surface area of what is being asserted against, I believe these changes better reflect what the test is trying to verify.

### Automated test coverage

Modified existing tests

### QA Plan

No QA

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

**Review by esoergel — COMMENTED** [review]
I like the idea in general, but I'm not certain these tests _are_ testing narrowly those specific fields.  Like, you may have an import that excludes a specific field, and the old assertion verifies that those fields are still present and unchanged after the import, or one that demonstrates what happens when a field is set to blank, or conflicts/agrees with the profile, and so forth.  It's basically "I'm testing a specific set of operations on user data and want to verify the final state in its totality."

That said, I do think you could certainly write a test suite that tests all of these things in isolation, I'm just not familiar enough with this one to be confident that limiting these assertions still preserves the meaningful scenarios.  If you are confident that that's the case, that's good enough for me.

**Comment by mjriley** [comment]
> I like the idea in general, but I'm not certain these tests _are_ testing narrowly those specific fields. 

Tests should build on each other. We definitely want to thoroughly test creating a new user, as we need to make sure that the entire user is as we expect. And there is an argument to do that same kind of testing on updating an existing user, as it's realistic to expect we could make logic that would remove previously stored values when updating a single one. But, once those cases are covered, it does not make sense to me to test against the entire object when we've already done so. Here are the individual updates:

* `_test_user_data`: This is really multiple tests cases in one. The cases to test insert and update I left unchanged. However, there are cases to verify that an empty string or a falsy value can be assigned to a field. Because we've already verified user creation, we should just be verifying that the empty string/falsy value is recorded correctly
* `_test_user_data_profile_redundant`: tests that when a user is assigned the same data as present on the assigned profile, the user's data shows the field, but it is stored on the profile, not the user. Could instead assert against the profile id and verify that the redundant value isn't stored on the user object, but the rest of the fields aren't needed.
* `_test_user_data_profile_blank`: tests that a user with a set value that matches a profile field can be added, provided the value is empty. Again, we're leaning on the user creation test to account for the rest of the fields functioning correctly. 
* `_test_required_field_optional_if_profile_set`: tests that a blank, but required field does not cause an error provided the profile specifies it. We only need to verify that the user is created and that the required field is the profile's value
* `_test_profile_cant_overwrite_existing_data`: the second assertion essentially tests the same thing as `_test_required_field_optional_if_profile_set`
* `_test_uncategorized_data`: tests that uncategorized data is assigned to top-level fields and can be updated. The creation and update tests cover the other fields.

As written, I'm comfortable that these changes do not alter the meaning of the tests. If the tests were verifying against the entire object to account for some other edge case, I think separate tests should be added.

File: corehq/apps/user_importer/tests/test_importer.py [hunk 1]
```diff
@@ -101,6 +101,25 @@ def setup_locations(self):
     def assert_user_data_equals(self, expected):
         self.assertEqual(self.user.get_user_data(self.domain.name).to_dict(), expected)
 
+    def assert_user_data_contains(self, expected):
```
[the remark this answers is not in the document]
Comment (mjriley) [reply corehq/apps/user_importer/tests/test_importer.py:104]: Looks like this is deprecated: https://docs.python.org/2/library/unittest.html#unittest.TestCase.assertDictContainsSubset

[the remark this answers is not in the document]
Comment (esoergel) [reply corehq/apps/user_importer/tests/test_importer.py:104]: Ah bummer - I wasn't aware of that

**Review by esoergel — APPROVED** [review]
Thanks for the diligence, that's the sort of analysis I was thinking of, and seeing that you've thought through it in that regard I don't have any reservations about these changes.

**Pull request merged**